### PR TITLE
splash screen webpage

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers pretty simple one

### CODE REVIEW

File: desktop/app/splash.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+'use strict'
+
+// Checks to see if we've launched before (we touch started.txt). If not, write the file and open a browser
+
+const helpURL = 'https://keybase.io/docs/cli_kbstage'
+const filePath = 'started.txt'
```
Comment (cjb) [inline desktop/app/splash.js]: This makes the path pretty non-deterministic, I think -- it's the directory you happened to be inside when you ran Electron?  What's going to happen when we launch via double clicking on an icon in Applications etc?

Using the same dir we put the keybase.leveldb in or similar sounds good to me?

Comment (chrisnojima) [inline desktop/app/splash.js]: won't this be packaged and therefore it'll be inside the app bundle?

Comment (cjb) [inline desktop/app/splash.js]: I'm not sure what'll happen on OS X; that sounds plausible.

But on Linux someone might launch it by typing "keybase" from a shell in their home dir instead, and then I think they'd get $HOME/started.txt.  Or, if it uses the app's own directory on Linux, we'd fail to write the file because the package will have been installed as root, and their user won't have permission to write there.

File: desktop/app/main.js [hunk 2]
```diff
@@ -37,3 +39,5 @@ if (process.platform === 'darwin') {
 
 ipc.on('showMain', () => { mainWindow.show() })
 ipc.on('showTracker', () => { trackerWindow.show() })
+
+splash()
```
Comment (chrisnojima) [inline desktop/app/main.js]: launch splash on startup

File: desktop/app/splash.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+'use strict'
+
+// Checks to see if we've launched before (we touch started.txt). If not, write the file and open a browser
+import {socketRoot} from '../../react-native/react/constants/platform.native.desktop'
+import path from 'path'
+
+const helpURL = 'https://keybase.io/docs/cli_kbstage'
+const filePath = path.join(socketRoot, 'started.txt')
```
Comment (cjb) [inline desktop/app/splash.js]: Don't we want dataRoot instead of socketRoot?

File: desktop/app/splash.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+'use strict'
+
+// Checks to see if we've launched before (we touch started.txt). If not, write the file and open a browser
+import {socketRoot} from '../../react-native/react/constants/platform.native.desktop'
```
Comment (chrisnojima) [inline desktop/app/splash.js]: we now put this file by the socket file. We had a discussion @cjb @maxtaco , on startup this location should be setup by the daemon (the leveldb / data folder may not be). Currently these folders are in 2 different places. On linux the socketfile is in an ephemeral folder so this won't work until that's changed. Core ticket created: https://keybase.atlassian.net/browse/CORE-2123

File: desktop/app/splash.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import {shell} from 'electron'
+
+export default () => {
+  fs.stat(filePath, (err, stats) => {
+    if (err) {
+      try {
+        fs.writeFileSync(filePath, 'hello world')
+      }
+      catch (e) {
+        console.error(`Couldn't mark splash screen shown: ${e}`)
```
Comment (chrisnojima) [inline desktop/app/splash.js]: if we can't write it, let's not show the splash else we could easily get into a state where we show over and over which would be lame

File: react-native/react/constants/platform.native.desktop.js [hunk 6]
```diff
@@ -1,7 +1,39 @@
 'use strict'
 
 import {OS_DESKTOP} from './platform.shared'
+import path from 'path'
```
Comment (chrisnojima) [inline react-native/react/constants/platform.native.desktop.js]: moving more platform specific stuff into here so its in one place

File: react-native/react/engine/transport.desktop.js [hunk 7]
```diff
@@ -2,35 +2,20 @@
...
-    paths.map(path => {
-      let exists = fs.existsSync(path)
-      if (exists) {
-        console.log('Found keybased socket file at ' + path)
-        sockfile = path
-      }
-    })
-    if (!sockfile) {
+
+    let exists = fs.existsSync(socketPath)
```
Comment (chrisnojima) [inline react-native/react/engine/transport.desktop.js]: Instead of looping over the various platform files we just look in one place

**Comment by chrisnojima** [comment]
@cjb this got more complex, take another look please

File: react-native/react/constants/platform.native.desktop.js [hunk 8]
```diff
@@ -1,7 +1,39 @@
...
+const envedPathOSX = {
+  staging: 'KeybaseStaging',
+  devel: 'KeybaseDevel',
+  prod: 'Keybase'
+}
+
+function findSocketRoot () {
+  const paths = {
+    'darwin': `${process.env.HOME}/Library/Caches/${envedPathOSX[runMode]}/`,
+    'linux': `${process.env.XDG_RUNTIME_DIR}/keybase.${runMode}/`
```
Comment (cjb) [inline react-native/react/constants/platform.native.desktop.js]: Hm, even after the core change, this won't work, because the core change won't affect where $XDG_RUNTIME_DIR points to, and won't make $XDG_RUNTIME_DIR non-ephemeral.  What's the plan for this code?

Comment (cjb) [inline react-native/react/constants/platform.native.desktop.js]: @chrisnojima came over to me.  Plan:

* core will move socket to XDG_DATA_DIR
* then we'll have another patch against this code to use XDG_DATA_DIR instead of XDG_RUNTIME_DIR

:thumbsup:

**Comment by chrisnojima** [comment]
Just so this conversation is captured. Spoke to @cjb in person.

- On startup at this point in the code only the socketfile directory is made. I don't want the ui to be building paths and setting permissions and etc so it has to be somehow associated with this folder, so yes this is supposed to use socketRoot
- Right now on linux socketRoot is ephemeral storage, that needs to change (or this needs to get more complicated). There doesn't seem to be a real reason for it to be in ephemeral storage so fixing that will fix this too.

tldr; This PR will show the splash screen if you reboot on linux until we move the dir, but logically its working. Once CORE changes we'll need to move our pointer to the socketfile anyways, which'll make splash work too
